src/component/mxgraph/overlay/shapes.ts

```diff
@@ -15,16 +15,63 @@
  */
 import { mxgraph } from '../initializer';
 import { mxRectangle } from 'mxgraph';
+import { MxGraphCustomOverlayStyle } from './custom-overlay';
 
 // TODO remove ts-ignore when typed-mxgraph definitions will declare the mxText missing properties
 export class OverlayBadgeShape extends mxgraph.mxText {
-  public constructor(value: string, bounds: mxRectangle) {
-    super(value, bounds);
+  public constructor(value: string, bounds: mxRectangle, style: MxGraphCustomOverlayStyle) {
+    super(
+      value,
+      bounds,
+      // align
+      undefined,
+      // valign
+      undefined,
+      style.font?.color,
+      // family
+      undefined,
+      style.font?.size,
+      // fontStyle
+      undefined,
+      // spacing
+      undefined,
+      // spacingTop
+      undefined,
+      //  spacingRight
+      undefined,
+      //   spacingBottom
+      undefined,
+      //   spacingLeft
+      undefined,
+      //  horizontal
+      undefined,
+      style.fill?.color,
+      style.stroke?.color,
+      //   wrap
+      undefined,
```

Comment:
⚠️ I suggest to remove the undefined trailing parameter. We can keep the comment if we want to describe what are the remaing parameters that we currently don't use.

Sonar also complains about using undefined values for optional parameters.

![image](https://user-images.githubusercontent.com/27200110/114352735-7ebcae80-9b6c-11eb-9458-35eaf96df576.png)